fallback.c

```diff
@@ -426,15 +429,24 @@ find_boot_option(EFI_DEVICE_PATH *dp, EFI_DEVICE_PATH *fulldp,
 		return EFI_OUT_OF_RESOURCES;
 	}
 
-	varname[0] = 0;
 	while (1) {
-		UINTN varname_size = sizeof(varname);
+		UINTN varname_size = buffer_size;
 		efi_status = gRT->GetNextVariableName(&varname_size, varname,
 						      &vendor_guid);
 		if (EFI_ERROR(efi_status)) {
-			if (efi_status == EFI_BUFFER_TOO_SMALL)
+			if (efi_status == EFI_BUFFER_TOO_SMALL) {
 				VerbosePrint(L"Buffer too small for next var "
-					     L"name\n");
+					     L"name, re-allocating it to be %d"
+					     L" chars long and retrying\n",
```

Comment:
Don't break the string across lines here (even though it was incorrectly broken before.)